### ISSUE TITLE
storage: useless rebalancing retries

### ISSUE THREAD

**Issue by petermattis** [cockroachdb/cockroach#15370]
`replicateQueue` [has a retry loop](https://github.com/cockroachdb/cockroach/blob/master/pkg/storage/replicate_queue.go#L225-L244) for operations. The comment in the code indicates failed preemptive snapshots are retried under the expectation that the rebalance operation will try to send the snapshot to a different node. But from the logs below, taken from `blue`, we can see that we're trying to send to the same node repeatedly.

```
I170426 18:40:17.172566 121 storage/replicate_queue.go:234  [replicate,n1,s1,r15/1:/Table/51/1/8{068085…-645563…}] snapshot failed: (n4,s4):?: remote declined snapshot: reservation rejected
I170426 18:40:17.230587 121 storage/replica_raftstorage.go:415  [replicate,n1,s1,r15/1:/Table/51/1/8{068085…-645563…}] generated preemptive snapshot c0baf247 at index 418875
I170426 18:40:17.233793 121 storage/replicate_queue.go:234  [replicate,n1,s1,r15/1:/Table/51/1/8{068085…-645563…}] snapshot failed: (n4,s4):?: remote declined snapshot: reservation rejected
I170426 18:40:17.342716 121 storage/replica_raftstorage.go:415  [replicate,n1,s1,r15/1:/Table/51/1/8{068085…-645563…}] generated preemptive snapshot 48ffa806 at index 418902
I170426 18:40:17.345584 121 storage/replicate_queue.go:234  [replicate,n1,s1,r15/1:/Table/51/1/8{068085…-645563…}] snapshot failed: (n4,s4):?: remote declined snapshot: reservation rejected
I170426 18:40:17.537053 121 storage/replica_raftstorage.go:415  [replicate,n1,s1,r15/1:/Table/51/1/8{068085…-645563…}] generated preemptive snapshot 302b777b at index 418947
I170426 18:40:17.546181 121 storage/replicate_queue.go:234  [replicate,n1,s1,r15/1:/Table/51/1/8{068085…-645563…}] snapshot failed: (n4,s4):?: remote declined snapshot: reservation rejected
I170426 18:40:17.977454 121 storage/replica_raftstorage.go:415  [replicate,n1,s1,r15/1:/Table/51/1/8{068085…-645563…}] generated preemptive snapshot e69eed75 at index 419075
I170426 18:40:17.978652 121 storage/replicate_queue.go:234  [replicate,n1,s1,r15/1:/Table/51/1/8{068085…-645563…}] snapshot failed: (n4,s4):?: remote declined snapshot: reservation rejected
I170426 18:40:18.705476 121 storage/replica_raftstorage.go:415  [replicate,n1,s1,r15/1:/Table/51/1/8{068085…-645563…}] generated preemptive snapshot c2c6fe50 at index 419282
I170426 18:40:18.708713 121 storage/replicate_queue.go:234  [replicate,n1,s1,r15/1:/Table/51/1/8{068085…-645563…}] snapshot failed: (n4,s4):?: remote declined snapshot: reservation rejected
E170426 18:40:18.708758 121 storage/queue.go:634  [replicate,n1,s1,r15/1:/Table/51/1/8{068085…-645563…}] failed to replicate after 5 retries
```

**Comment by a-robinson** [comment]
Just a +1 that this was **really** annoying when looking at blue the other day.

**Comment by petermattis** [comment]
Tried fixing this issue by not retrying if we get the same error twice in a row. For some reason I'm unable to reproduce the problem right now. Probably just jetlag.

**Comment by petermattis** [comment]
Not jet lag. https://github.com/cockroachdb/cockroach/pull/15619 flipped the values for the declined and failed reservation timeouts. @BramGruneir was this intentional?

**Comment by BramGruneir** [comment]
No it was not!  I'll send a fix right away.

**Comment by petermattis** [comment]
@BramGruneir Ok. I'm in this code at the moment. I'll fix it.

**Comment by BramGruneir** [comment]
already done, and I'd like to cherry pick this fix to 1.0.2
